aql/src/main/java/org/ehrbase/openehr/sdk/aql/parser/AqlQueryVisitor.java

```diff
@@ -710,7 +742,19 @@ public OrderByExpression visitOrderByExpr(OrderByExprContext ctx) {
         return orderByExpression;
     }
 
-    private static String unescapeText(RuleContext ctx) {
+    @Override
+    public TerminologyFunction visitTerminologyFunction(AqlParser.TerminologyFunctionContext ctx) {
+
+        TerminologyFunction terminologyFunction = new TerminologyFunction();
+
+        terminologyFunction.setOperation(unescapeText(ctx.STRING(0).getText()));
+        terminologyFunction.setServiceApi(unescapeText(ctx.STRING(1).getText()));
+        terminologyFunction.setUriParameters(unescapeText(ctx.STRING(2).getText()));
+
+        return terminologyFunction;
+    }
+
+    private static String unescapeText(String ctxText) {
```

Comment:
better ParseTree as parameter type

Reply:
`    public void setPredicates(List<AndOperatorPredicate> predicates) {
        if (predicates != null) {
            this.predicates = new ArrayList<>(predicates);
        } else {
            this.predicates = null;
        }
    }`

Reply:
I guess the response was meant for a different comment?

Reply:
yes